.github/workflows/profiler-pipeline.yml

```diff
@@ -0,0 +1,653 @@
+          #
+          echo "DD_TESTING_PROFILER_FOLDER=$(realpath .)/DDProf-Deploy" >> $GITHUB_ENV
+          # create
+          tests_output_dir=$(realpath .)/tests_output
+          mkdir $tests_output_dir
+          echo "DD_TESTING_OUPUT_DIR=$tests_output_dir" >> $GITHUB_ENV
+
+      - name: Run Integration tests
+        env:
+          DD_API_KEY: '${{ secrets.DD_API_KEY }}'
+        run: |
+          cd profiler/test/Datadog.Profiler.IntegrationTests &&
+          LD_PRELOAD=${DD_TESTING_PROFILER_FOLDER}/Datadog.Linux.ApiWrapper.x64.so dotnet test -c Release -p:Platform=x64 --logger "trx" -- RunConfiguration.TreatNoTestsAsError=true
+
+      - name: Generate tests report
+        uses: dorny/test-reporter@v1
+        if: '${{ always() }}'
+        with:
+          name: STests Linux x64 report
+          path: profiler/test/Datadog.Profiler.IntegrationTests/TestResults/*.trx
+          reporter: dotnet-trx
+
+      - name: 'Publish Tests result, Logs and PProf files'
+        uses: actions/upload-artifact@v2
+        if: '${{ always() }}'
+        with:
+          if-no-files-found: error
+          name: Tests_result_logs_and_pprofs_files.Linux.Release.x64
+          path: |
+            ${{ env.DD_TESTING_OUPUT_DIR }}
+          retention-days: 7
+
+  benchmark_linux_x64:
+    name: BMmark Linux x64
+    runs-on: ubuntu-latest
+    needs: build_linux_release_x64
+    strategy:
+      fail-fast: false
+      matrix:
+        framework:
+          - netcoreapp31
+          - net50
+    services:
+      dd-agent:
+        image: 'datadog/agent:latest'
+        ports:
+          - '8126:8126'
+        env:
+          DD_API_KEY: '${{ secrets.DD_API_KEY }}'
+          DD_INSIDE_CI: true
+          DD_HOSTNAME: dd-profiler-dotnet
+          DD_SERVICE: dd-profiler-dotnet
+
+    steps:
+      - name: Clone dd-trace-dotnet repository
+        uses: actions/checkout@v2
+
+      - name: Download profiler artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: DDProf-Deploy.Linux.Release.x64
+          path: DDProf-Deploy
+
+      - name: Setup Go 1.16.8
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16.8
+
+      - name: Prepare environment to run the application
+        run: |
+          #
+          # set profiler deployment and test output dir folder path for the test
+          #
+          go_tool_dir=$(realpath .)/tools
+          mkdir $go_tool_dir
+          echo "GOPATH=$go_tool_dir" >> $GITHUB_ENV
+
+      - name: Install timeit
+        run: go install github.com/tonyredondo/timeit@latest
+
+      - name: Build sample applications
+        run: dotnet build -c Release profiler/src/Demos/Datadog.Demos.sln -p:Platform=x64
+
+      - name: Execute PiComputation benchmark
+        run: |
+          cd profiler/build
+          ./run.sh ${{ matrix.framework }}
+
+      - name: Wait 20 seconds to agent flush before finishing pipeline
+        run: sleep 20
+
+  build_windows:
```

Comment:
Should this stage be renamed to make it clearer that it's profiling? As of now, it does not conflicts with us as we have a build_windows_tracer and build_windows_profiler stages.
But we could maybe prepend all stages here with cp_ to make it crystal clear? WDYT?

Also, on our end I assume we should change our build_windows_profiler step to build it from the repo now.

Reply:
There won't be any conflict since it's the name below `name:` which is used a the key.
I do thing you have a `Build Windows` ( with the space) step.
We could have  `Build Profiler Windows` instead.

> Also, on our end I assume we should change our build_windows_profiler step to build it from the repo now.
I already done that in the previous PR.